### DISCUSSION TITLE
Test Vendor ID and Device Attestation Certificate/Network Commissioning cluster

### DISCUSSION THREAD

**Discussion by javierbarellano4v** [matter-js/matter.js#69] [Q&A]
Just to trying to get clarification.

I'm assuming that node-matter is using of one test-vendor IDs by default.

Furthermore, with that assumption, d**oes node-matter have a device attestation certificate**? 

My understanding is that a cert is normally stored on a Matter device.

Secondly, is the Network Commissioning cluster implemented in node-matter which includes the **AddOrUpdateWiFiNetwork** command?

Thanks.

**Comment by Apollon77** [comment]
Yes normally the DAC is provided after certification has been approved and is then "static" on the device.

The current main branch version will generate the Device Attestation Certification and also all other certificates "On the fly" based on the vendor-id/product-id provided, but the base (means PAA aka Authority and the Signing key for the DAC) are the official ones from Matter test. But all these data can manually be overridden in the code and this will be easier in the future too, but for "having a dev/test device this is the easiest way.

For your other question: No, AddOrUpdateWiFiNetwork is not yet implemented in the Network commissioning cluster because till now we foxus on Ethernet connectivity.
Contributions are welcome :-)

**Comment by javierbarellano4v** [comment]
@Apollon77 Appreciate the detailed explanation, I figured that there some overriding or some 'custom' setup to create the certs. 

I had previously asked about 'Bluetooth beaconing' but not on the roadmap yet - brings in dependencies as well.

**Comment by javierbarellano4v** [comment]
just to go further....

**If** node-matter **did** (or when it does) implement Bluetooth and the Networking Commissioning cluster with AddOrUpdateWifiNetwork, when there is an attempt to commission it with, e.g., Amazon Alexa, should you expect for Alexa, to automatically attempt to execute the AddOrUpdateWifiNetwork command and set the Wi-Fi credentials on nodematter?

**Reply by Apollon77** [reply]
I would expect so ... but all in all this then adds even more interesting challenges because wifi connections are usually handled by the "host os" and not by nodejs. So this would even add another layer to the topic which makes it even more "specific". So most likely big parts of this logic then needs to be implemented by the dev itself anyway and the library can only "support" there. 

We need to see how we arrange all this when we come to it.

**Comment by javierbarellano4v** [comment]
i.e., library supports interfaces, etc.. and the dev implements the details (logic as you say) of actually setting Wi-Fi connection.

**Reply by Apollon77** [reply]
Exactly
